### DISCUSSION TITLE
Remember window position per display

### DISCUSSION THREAD

**Discussion by odnoletkov** [rxhanson/Rectangle#400] [Ideas]
Currently when window is sent to a different display it is positioned in the center of this new display (and size is adjusted to fit window when needed). This is a good default behavior for the initial position. But if the user already positioned *this* window on *this* display before it would be great to restore this previously known position.

I tend to move some windows/apps between screens often – when switching between contexts. Currently I need to re-position each window manually.

It looks like this can be built using existing `Restore` logic, extending it to have separate history per each screen?

What do you think about the idea? If you like it but don't have time to implement, I can try to submit a PR

**Comment by rxhanson** [comment]
This idea makes sense to me. I can also see it being a little strange as the default behavior of the app, though, for users that don't consistently put windows in the same locations every time on a specific monitor. I don't think I'll implement this one myself, but I'm pretty sure I'd accept a pull request for it, being that it would be configurable via terminal command. If it's in and seems like it should be the default behavior, we could always make it that in the future.

Here's a couple things to keep in mind. 
Rectangle doesn't keep track of displays. Every single shortcut execution checks which displays are available. This isn't a big deal, we'd just have to keep track of the displays somewhere.
Keeping a history per display might not be necessary, just the last size/position on that display.

**Comment by sstarcher** [comment]
This currently bites me when unplugging my MacBook and then going back to closing the MacBook and reconnecting it to the display.
